### PR TITLE
feat: Added limits to attributes

### PR DESCRIPTION
**Pull request by Ashutosh619-sudo**
### Description
Please add PR description here, add screenshots if needed

## Clickup
Please add link here
https://app.clickup.com/1864988/v/l/6-901603904304-1

## Summary by CodeRabbit

- **New Features**
	- Introduced a new method to enforce synchronization limits for various entity types.
	- Updated synchronization processes to include checks for permitted sync operations based on defined limits.

- **Bug Fixes**
	- Enhanced the synchronization process to prevent exceeding attribute limits, improving overall reliability.

- **Documentation**
	- Added comments for better clarity and understanding of the synchronization logic.

- **Tests**
	- Expanded test coverage for synchronization logic, including enhancements for expense attributes, employee mappings, and error resolution.
	- Added tests to validate the skipping of synchronization when limits are exceeded.

### CODE REVIEW

File: apps/sage_intacct/utils.py [hunk 1]
```diff
@@ -97,7 +105,20 @@ def get_tax_exclusive_amount(self, amount, default_tax_code_id):
...
+    
 
+    def is_sync_allowed(self, attribute_type: str, attribute_count: int):
+        """
+        Checks if the sync is allowed
+
+        Returns:
+            bool: True
+        """
+        workspace_created_at = Workspace.objects.get(id=self.workspace_id).created_at
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:117]:
```

if attribute_count > SYNC_UPPER_LIMIT[attribute_type]:
    workspace_created_at = Workspace.objects.get(id=self.workspace_id).created_at
    if workspace_created_at > timezone.make_aware(datetime(2024, 10, 1), timezone.get_current_timezone()):
        return False
    else:
        return True

return True
```
output doesn't change but this queries db only when count is > limit

File: apps/sage_intacct/utils.py [hunk 2]
```diff
@@ -312,76 +352,87 @@ def sync_projects(self):
...
+                project_attributes.append({
+                    'attribute_type': 'PROJECT',
+                    'display_name': 'project',
+                    'value': project['NAME'],
+                    'destination_id': project['PROJECTID'],
+                    'active': project['STATUS'] == 'active',
+                    'detail': detail
+                })
+        
+        DestinationAttribute.bulk_create_or_update_destination_attributes(
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:384]: upsert should happen per page

File: apps/sage_intacct/utils.py [hunk 3]
```diff
@@ -312,76 +352,87 @@ def sync_projects(self):
...
+                        'attribute_type': 'ITEM',
+                        'display_name': 'item',
+                        'value': item['NAME'],
+                        'destination_id': item['ITEMID'],
+                        'active': True
+                    })
 
-            DestinationAttribute.bulk_create_or_update_destination_attributes(
-                item_attributes, 'ITEM', self.workspace_id, True)
+        DestinationAttribute.bulk_create_or_update_destination_attributes(
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:422]: upsert should happen per page

File: apps/sage_intacct/utils.py [hunk 4]
```diff
@@ -595,57 +646,68 @@ def sync_classes(self):
...
+                    'attribute_type': 'CLASS',
+                    'display_name': 'class',
+                    'value': _class['NAME'],
+                    'destination_id': _class['CLASSID'],
+                    'active': True
+                })
 
-            DestinationAttribute.bulk_create_or_update_destination_attributes(
-                class_attributes, 'CLASS', self.workspace_id, True)
+        DestinationAttribute.bulk_create_or_update_destination_attributes(
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:668]: same as above

File: apps/sage_intacct/utils.py [hunk 5]
```diff
@@ -595,57 +646,68 @@ def sync_classes(self):
...
+                    'attribute_type': 'CUSTOMER',
+                    'display_name': 'customer',
+                    'value': customer['NAME'],
+                    'destination_id': customer['CUSTOMERID'],
+                    'active': True
+                })
 
-            DestinationAttribute.bulk_create_or_update_destination_attributes(
-                customer_attributes, 'CUSTOMER', self.workspace_id, True)
+        DestinationAttribute.bulk_create_or_update_destination_attributes(
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:697]: same

File: tests/test_sageintacct/test_utils.py [hunk 6]
```diff
@@ -459,6 +473,10 @@ def test_sync_tax_details(mocker, db):
 def tests_sync_accounts(mocker, db):
     workspace_id = 1
 
+    mocker.patch(
```
Comment (ashwin1111) [inline tests/test_sageintacct/test_utils.py:476]: pls to add tests for skip case by mocking and returning high values

File: tests/test_sageintacct/test_utils.py [hunk 7]
```diff
@@ -930,3 +931,39 @@ def mock_allocations_generator(field, value, updated_at=None):
             assert set(allocation_attribute.detail.keys()) == {'LOCATIONID'}
         else:
             assert set(allocation_attribute.detail.keys()) == {'LOCATIONID', 'GLDIMWHAT_IS_NILESH_PANT'}
+
+
+def test_skip_sync_attributes(mocker, db):
+    workspace_id = 1
+
+    mocker.patch(
+        'sageintacctsdk.apis.Classes.count',
```
Comment (ashwin1111) [inline tests/test_sageintacct/test_utils.py:940]: same comment as netsuite

**Review by ashwin1111 — APPROVED** [review]

**Review by ruuushhh — APPROVED** [review]

**Pull request merged**